pkg/vcdsdk/gateway.go

```diff
@@ -1949,3 +1950,40 @@ func (gm *GatewayManager) UpdateLoadBalancer(ctx context.Context, lbPoolName str
 	}
 	return vsSummary.VirtualIpAddress, nil
 }
+
+func (gm *GatewayManager) FetchIpSpacesBackingGateway(ctx context.Context) ([]string, error) {
+	ipSpaceService := gm.Client.APIClient.IpSpacesApi
+
+	filterString := fmt.Sprintf("gatewayId==%s", gm.GatewayRef.Id)
+	options := swaggerClient.IpSpacesApiGetFloatingIpSuggestionsOpts{Filter: optional.NewString(filterString), SortAsc: optional.NewString("ipSpaceRef.name"), SortDesc: optional.EmptyString()}
+	var pageSize int32 = 10
+	var pageNum int32 = 1
+	var resultTotal int32 = math.MaxInt32
+
+	var ipSpaceIds []string
+	for int32(math.Ceil(float64(resultTotal)/float64(pageSize))) >= pageNum {
+		suggestions, _, err := ipSpaceService.GetFloatingIpSuggestions(ctx, pageNum, pageSize, &options)
+		if err != nil {
+			return nil, err
+		}
+		resultTotal = suggestions.ResultTotal
+
+		if suggestions.Values != nil {
```

Comment:
Style nit: when we have blocks with returns, it is better to have the return (error-handling) in the block and perform the default logic outside.

So we would have:
```
if suggestions.Values == nil {
    return nil, fmt.Errorf(...)
}
// regular break, loop etc outside the else part of the `if`
```